### PR TITLE
POC implementation of SASL for SMTP

### PR DESCRIPTION
**Pull request by chibenwa**
The very beginning of something...

### CODE REVIEW

**Comment by chibenwa** [comment]
```
$ openssl s_client -connect 172.17.0.2:465
CONNECTED(00000003)
Can't use SSL_get_servername
depth=0 C = FR, ST = Unknown, L = Puteaux, O = James, OU = Linagora, CN = Benoit Tellier
verify error:num=18:self signed certificate
verify return:1
depth=0 C = FR, ST = Unknown, L = Puteaux, O = James, OU = Linagora, CN = Benoit Tellier
verify error:num=10:certificate has expired
notAfter=Nov 24 07:32:55 2015 GMT
verify return:1
depth=0 C = FR, ST = Unknown, L = Puteaux, O = James, OU = Linagora, CN = Benoit Tellier
notAfter=Nov 24 07:32:55 2015 GMT
verify return:1
---
Certificate chain
 0 s:C = FR, ST = Unknown, L = Puteaux, O = James, OU = Linagora, CN = Benoit Tellier
   i:C = FR, ST = Unknown, L = Puteaux, O = James, OU = Linagora, CN = Benoit Tellier
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIDeTCCAmGgAwIBAgIEXaLC/zANBgkqhkiG9w0BAQsFADBtMQswCQYDVQQGEwJG
UjEQMA4GA1UECBMHVW5rbm93bjEQMA4GA1UEBxMHUHV0ZWF1eDEOMAwGA1UEChMF
SmFtZXMxETAPBgNVBAsTCExpbmFnb3JhMRcwFQYDVQQDEw5CZW5vaXQgVGVsbGll
cjAeFw0xNTA4MjYwNzMyNTVaFw0xNTExMjQwNzMyNTVaMG0xCzAJBgNVBAYTAkZS
MRAwDgYDVQQIEwdVbmtub3duMRAwDgYDVQQHEwdQdXRlYXV4MQ4wDAYDVQQKEwVK
YW1lczERMA8GA1UECxMITGluYWdvcmExFzAVBgNVBAMTDkJlbm9pdCBUZWxsaWVy
MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAhW0GWSgkK44XQpwLn8KX
q9kJ3zgDYllEX7W8p+3sBYMAP0JC+lzh42mX/8XHti0vkmv/vjk0paAsB9s5uYhv
/W94sdqjexi3L213+OD5Kcy+2tTgXN1ucF65d/dDeqGHAyBs1rm6LgyRwQ5ye8Cl
OtXbSkz5qujmnFDDzrDIgzefxFsTHHIBTsdyttq2Atqzgdg2LHaFbIntwr6lfl9v
puXr8p+CY6PfehX8mdmaJ7J/gfAll3zzdgeNUoAW0eDvcsphJ06elsDahieo0n/o
XVhy+TGWtAQRDZQDjeK2BX/vVSPAlGOW7hVTv7WycAolzKlXZyiJMnwTWJl6YoUP
vQIDAQABoyEwHzAdBgNVHQ4EFgQU7tDLswRlOJqoX4aVgrXRQbmOYbIwDQYJKoZI
hvcNAQELBQADggEBAGmzK2i2H5D6xITyN1iNQhXbABQ/3rz9K5dEvy0ZLgYOUqyw
1WD43Do/A3eZuEZbuuYAlbvDEUNgS7rJ5HCrwIKFbPHWmfRX2j59UX+R8fI6G9wB
qqcRYWpert347rGT/7MFLHk55LL6Tf//fwoWu6GWRj3wTvosVuunUiiC6zTS4MN9
moJp+IF03Q6JOPWu7/tfaKfXQHxG/hK492oV2vBG2r29UUJW6YO1S0DK+/cU0cCu
+jqbY1ZOIAk906onRUFoPGuypOm3vmbE6mo5o49rNtp+VmZagZ7GsjJ4KWJB1c6d
SFNIFlH2VlS8Lywr2tesClWO5tqtMswRsoA9GeU=
-----END CERTIFICATE-----
subject=C = FR, ST = Unknown, L = Puteaux, O = James, OU = Linagora, CN = Benoit Tellier

issuer=C = FR, ST = Unknown, L = Puteaux, O = James, OU = Linagora, CN = Benoit Tellier

---
No client certificate CA names sent
Peer signing digest: SHA256
Peer signature type: RSA-PSS
Server Temp Key: X25519, 253 bits
---
SSL handshake has read 1425 bytes and written 363 bytes
Verification error: certificate has expired
---
New, TLSv1.3, Cipher is TLS_AES_256_GCM_SHA384
Server public key is 2048 bit
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 10 (certificate has expired)
---
---
Post-Handshake New Session Ticket arrived:
SSL-Session:
    Protocol  : TLSv1.3
    Cipher    : TLS_AES_256_GCM_SHA384
    Session-ID: 9E01E908FF479ABE914CA9904DFFB84064917E74BDBFB4A9644FE501DF1FE897
    Session-ID-ctx: 
    Resumption PSK: 688B7E6893AB53B69DC9107E6AFBDB39C7116A4E71B01EB902E0813334BC99275AC59D0CABC8FEF37D9B3E8C07C2429E
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 86400 (seconds)
    TLS session ticket:
    0000 - 48 9f 54 db 2d bb da 00-80 1e 5d 4e 1d fd 56 dd   H.T.-.....]N..V.
    0010 - 16 97 2a d5 7e 72 f6 3a-a9 04 c8 94 5b f1 16 50   ..*.~r.:....[..P

    Start Time: 1639551801
    Timeout   : 7200 (sec)
    Verify return code: 10 (certificate has expired)
    Extended master secret: no
    Max Early Data: 0
---
read R BLOCK
220 Apache JAMES awesome SMTP Server
ehlo o
250-f4a187d3dc70 Hello o [172.17.0.1])
250-AUTH LOGIN PLAIN OAUTHBEARER
250-AUTH=LOGIN PLAIN OAUTHBEARER
250-PIPELINING
250-ENHANCEDSTATUSCODES
250 8BITMIME
AUTH OAUTHBEARER bix1c2VyPWZpcnN0bmFtZTE0LnN1cm5hbWUxNEB1cG4uaW50ZWdyYXRpb24tb3Blbi1wYWFzLm9yZwFhdXRoPWV5SmhiR2NpT2lKU1V6STFOaUlzSW5SNWNDSWdPaUFpU2xkVUlpd2lhMmxrSWlBNklDSjFSWE5XVjNoT2NrUXhVM0JKV1V4WlF6VTViV04wUTE5RVNITTFhRkZmTjBOMVRrZDNZakpTV2tWQkluMC5leUpsZUhBaU9qRTJNemsxTlRVeE1EZ3NJbWxoZENJNk1UWXpPVFUxTVRVd09Dd2lZWFYwYUY5MGFXMWxJam94TmpNNU5UVXhOVEEyTENKcWRHa2lPaUl4WWpWbE1HUmxNaTA0TWpOaExUUXhaRE10T1RVMk1pMHhNems1TjJVNVlXTm1NallpTENKcGMzTWlPaUpvZEhSd2N6b3ZMMkYxZEdndWRYQnVMbWx1ZEdWbmNtRjBhVzl1TFc5d1pXNHRjR0ZoY3k1dmNtY3ZZWFYwYUM5eVpXRnNiWE12ZFhCdUlpd2lZWFZrSWpvaVlXTmpiM1Z1ZENJc0luTjFZaUk2SWpJelpUQmxaamczTFRaaFlUTXRORGRrWVMxaFkyTmlMVEk1WXpVNE9HUXlZekZrT1NJc0luUjVjQ0k2SWtKbFlYSmxjaUlzSW1GNmNDSTZJbTl3Wlc1d1lXRnpJaXdpYzJWemMybHZibDl6ZEdGMFpTSTZJbVEzWXpJNU5qSm1MVFl5WW1FdE5EUTVZUzA0Wm1GakxUSTVZVFUyWkdKbU5qWmlNQ0lzSW1GamNpSTZJakVpTENKaGJHeHZkMlZrTFc5eWFXZHBibk1pT2xzaUtpSmRMQ0p5WldGc2JWOWhZMk5sYzNNaU9uc2ljbTlzWlhNaU9sc2liMlptYkdsdVpWOWhZMk5sYzNNaUxDSjFiV0ZmWVhWMGFHOXlhWHBoZEdsdmJpSXNJbVJsWm1GMWJIUXRjbTlzWlhNdGRYQnVJbDE5TENKeVpYTnZkWEpqWlY5aFkyTmxjM01pT25zaVlXTmpiM1Z1ZENJNmV5SnliMnhsY3lJNld5SnRZVzVoWjJVdFlXTmpiM1Z1ZENJc0ltMWhibUZuWlMxaFkyTnZkVzUwTFd4cGJtdHpJaXdpZG1sbGR5MXdjbTltYVd4bElsMTlmU3dpYzJOdmNHVWlPaUp2Y0dWdWFXUWdjSEp2Wm1sc1pTQmxiV0ZwYkNJc0ltVnRZV2xzWDNabGNtbG1hV1ZrSWpwbVlXeHpaU3dpYm1GdFpTSTZJa1pwY25OMGJtRnRaVEUwSUZOMWNtNWhiV1V4TkNJc0luQnlaV1psY25KbFpGOTFjMlZ5Ym1GdFpTSTZJbVpwY25OMGJtRnRaVEUwTG5OMWNtNWhiV1V4TkNJc0ltZHBkbVZ1WDI1aGJXVWlPaUpHYVhKemRHNWhiV1V4TkNJc0ltWmhiV2xzZVY5dVlXMWxJam9pVTNWeWJtRnRaVEUwSWl3aVpXMWhhV3dpT2lKbWFYSnpkRzVoYldVeE5DNXpkWEp1WVcxbE1UUkFkWEJ1TG1sdWRHVm5jbUYwYVc5dUxXOXdaVzR0Y0dGaGN5NXZjbWNpZlEuVmRBZ0o5T25MU2x5YTBnQzBBQW9QMjNmdjIwRXRvejczbEFHY05OTmoxT001ZUx1ZW5BTTdtMDZ1S0tMVHRYNHBxYklZUlMwSnJNYVNSV254Ty1CV25KZEUzS2hGNVNQR3p1VFpTdExvTW5RcE9aTVpLNXFsZWpEN0RhV2t5NkN2eUNjWFBEbW84ZXI3Y19kazZPR0tBWE9Qd1dkTHN5Q0JFRkwzQ0Uzd0RlZ0czWml2VGstOWtYY1RBSHhfLTQ3TUhPWGNhNnp3eWpyUlczZy1iZGhFek5xQ0VzZ1RzOHdvM3Y5c0l4RlZmb19SRlZWT3pPRzRRcFZKT0lGclR3Z0Z2M0ZLaWlKeXI3WWd2bHBKVGxQQzJGLUJuMHZ0TGVJdTFpSDVLTEVobGEzb3NwSWpTZXVzZUNscUFFLV8xemU3QlVxSkh5S1VycGMyYjVOODR3elFBAQE=
235 Authentication Successful
```

File: protocols/api/src/main/java/org/apache/james/protocols/api/OIDCSASLParser.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        public String getUser() {
+            return user;
+        }
+
+        public String getToken() {
+            return token;
+        }
+    }
+
+    public static Optional<OIDCInitialResponse> parse(String initialResponse) {
```
Comment (vttranlina) [inline protocols/api/src/main/java/org/apache/james/protocols/api/OIDCSASLParser.java:56]: What is `initialResponse` ?

Comment (chibenwa) [reply protocols/api/src/main/java/org/apache/james/protocols/api/OIDCSASLParser.java:56]: Syntax:

```
C: AUTH OAUTHBEARER initialClientResponse
```

CF RFC-7628

**Comment by chibenwa** [comment]
See https://github.com/apache/james-project/pull/810 instead

**Pull request closed without merging**